synapse/storage/databases/state/bg_updates.py

```diff
@@ -340,14 +360,14 @@ def reindex_txn(txn):
 
         return result * BATCH_SIZE_SCALE_FACTOR
 
-    async def _background_index_state(self, progress, batch_size):
-        def reindex_txn(conn):
+    async def _background_index_state(self, progress: dict, batch_size: int) -> int:
+        def reindex_txn(conn: LoggingDatabaseConnection) -> None:
             conn.rollback()
             if isinstance(self.database_engine, PostgresEngine):
                 # postgres insists on autocommit for the index
                 conn.set_session(autocommit=True)
                 try:
-                    txn = conn.cursor()
+                    txn = conn.LoggingTransaction()
```

Comment:
Huh? `LoggingDatabaseConnection` doesn't have such a method, and `cursor()` should be correct anyway?

Reply:
I have no idea what happened here...